Updates readme (#3)
Digital-Health-AI-Startup/frontend-test-template: README.md

## The Process

1. Start by using this template. You can do so by clicking the "Use this template" button in the top right of the GitHub UI, then "Create a new repository". Make this new repo private.
2. Read the details about the test below and complete the test.
3. When you are ready to submit, open two PRs as outlined below.
4. Invite the code review team to your repository and email us to let us know it's ready. If the list of GitHub users has not already been provided to you, work with your contact here to get that list.
5. We will review your work and schedule a technical interview where you will meet some of our engineers and discuss this test.

### Submitting the Test

When you are ready to submit the challenge, please open two PRs. One PR should consist of your solution for the **base challenge**. The second PR should consist of your solution to the **enhancement** you’ve selected.

In both of your PR descriptions, please include:

1. Relevant screenshots/screen-recordings.
2. A brief description of your submission.
3. Anything else you might like us to know as we review your submission.

> In the PR description for your enhancement, please state which enhancement you completed.

Author: This sentence was updated to remove the custom task reference

## The Test

### Background Information

For this challenge, let’s pretend that we have built a system that receives health data from a patient, like their heart rate and blood pressure, and will output “discoveries” about a patient. A discovery is a piece of data that represents an insight into a patient’s health. For example, if a patient has a resting heart rate of 60 beats per minute, and in a few weeks the patient’s resting heart rate increases to 65 bpm, a discovery might be generated that says, “Patient’s resting heart rate has increased from 60 bpm to 65 bpm.”

While our system does generate discoveries, these discoveries are not being displayed anywhere, therefore they cannot be viewed. Our clinicians have asked us to build an interface where discoveries can be viewed. Once a discovery has been reviewed by a clinician, the clinician should be able to move the discovery over to an “Acknowledged” list.

The “Acknowledged” list will signal to other clinicians that the discovery has been reviewed and validated by another clinician.

### The Assignment - What we want you to do

Here are the asks we have for you:

> **Base Challenge, Part One: Build an interface where these discoveries can be viewed by a user.**
> As a user, I want to be able to see the full list of 100 discoveries. The list of discoveries should look like what’s in the mock (DESIGN LINK HERE) . The discoveries should be listed in reverse-chronological order (most recent first).
> 
> **Base Challenge, Part Two: Create the Ability to “Acknowledge” Discoveries.**
> As a user, I want the ability to accept a single discovery, multiple discoveries, or all discoveries so that I can show that I have reviewed the discoveries that have been made for the patient.
> The user should be able to select one, multiple, or all discoveries.
> When a discovery is moved over to the “Acknowledged” list, a “Submit” button should appear.
> Clicking “Submit” should save the list of Acknowledged discoveries.
> After clicking “Submit,” the Acknowledged list should persist after a page refresh.

Reviewer: For this section i was thinking something like this? Feel free to edit or ignore this though. 

```suggestion
##### Acceptance Criteria  

1. As a user, I want to see the full list of 100 discoveries, displayed in reverse-chronological order (most recent first), and formatted according to the mock design (DESIGN LINK HERE).  
2. As a user, I want the ability to accept:  
   - A single discovery  
   - Multiple discoveries  
   - All discoveries  
   so that I can indicate which discoveries have been reviewed for the patient.  
3. As a user, I should be able to select one, multiple, or all discoveries.  
4. When a discovery is moved to the “Acknowledged” list:  
   - A “Submit” button should appear.  
   - Clicking “Submit” should save the list of acknowledged discoveries.  
   - The acknowledged discoveries list should persist after a page refresh.  
  
```

Author: I like this. I like framing this explicitly as AC, as opposed to just a user story with our asks. I also think separating it out into bullet points makes it easier to read and digest

I updated the readme to use ^^ your suggestion and pushed!

Author: FWIW Here's how that section looks now:
<img width="1009" alt="Screenshot 2025-03-20 at 2 11 22 PM" src="https://github.com/user-attachments/assets/13f3a24a-e3c8-4893-8c40-3c530a44c8ef" />

> # Take Home Challenge Readme

Reviewer 2: 🤷 
```suggestion
# Percipio Take Home Challenge
```

Author: done and pushed!

2. **We will only assess you for the code you submit that completes the base challenge and the one enhancement**. Please indicate in your submission which enhancement you completed.
3. **We expect you to spend 2-4 hours on this challenge**. If you’re unable to complete all the requirements in that time frame, feel free to explain in your write-up any of the thoughts or plans you would have completed with more time.
4. **We do not expect a full scale application with working navigation, complex animations, etc. You should only develop what you need in order to accomplish the goal we've given.**
5. The purpose of this challenge is to get a sense of your work, and to serve as the focal point of your technical interview. Expect the code you submit to come up in your technical interview.

## The Process

Before you start coding, make sure you have the following installed on your machine:

1. `nvm` or Node.js
2. `pnpm` (we recommend installing with Corepack)
3. (Optional) nx for running workspaces

When you are ready to start coding:

> 1. Clone this repo

Reviewer 2: ooh just realized this is a template repo, so we can pull over the existing instructions from our fullstack test:
```suggestion
1. Start by using this template. You can do so by clicking the "Use this template" button in the top right of the GitHub UI, then "Create a new repository". Make this new repo private.
```

Reviewer 2: ooh actually looking at those docs, I think most of that `Process` section would be good to include... (although we may want to run this by Eddie to ensure he's intending this same flow)
```
* Start by using this template. You can do so by clicking the "Use this template" button in the top right of the GitHub UI, then "Create a new repository". Make this new repo private.
* Read the details about the test below and complete the test.
* Invite the code review team to your repository and email us to let us know it's ready. If the list of GitHub users has not already been provided to you, work with your contact here to get that list.
* We will review your work and schedule a technical interview where you will meet some of our engineers and discuss this test.

Author: Updated this section to include ^^ those bullet points

2. Create your own branches on your new repo. Create one branch for the **base challenge**, and a second branch for the **enhancement**
3. When you are ready to submit, open a PR (more detailed instructions below)

Once you’ve got the repo set up, use nvm to install and path the correct Node version based on the .nvmrc in this repo:

```
nvm install
nvm use
```

Then, install packages from the monorepo root:

```
pnpm i
```

And then you can run both the API and web app together, concurrently, by running the following script from the root directory:

```
pnpm run start:dev
```

> You should be able to see the app on `http://localhost:5173/`.

Reviewer 2: maybe include both urls here? 🤷 

```suggestion
You should be able to see the app on `http://localhost:5173/` and the API documentation at `http://localhost:3000/api`.
```

Author: Good shout! Done and pushed

> There's more information about how to run the [API](./apps/api/README.md) and [UI](./apps/web/README.md) in their respective directories.

Author: Links to the other readmes

2. **We will only assess you for the code you submit that completes the base challenge and the one enhancement**. Please indicate in your submission which enhancement you completed.
3. **We expect you to spend 2-4 hours on this challenge**. If you’re unable to complete all the requirements in that time frame, feel free to explain in your write-up any of the thoughts or plans you would have completed with more time.
4. **We do not expect a full scale application with working navigation, complex animations, etc. You should only develop what you need in order to accomplish the goal we've given.**
5. The purpose of this challenge is to get a sense of your work, and to serve as the focal point of your technical interview. Expect the code you submit to come up in your technical interview.

## The Process

Before you start coding, make sure you have the following installed on your machine:

1. `nvm` or Node.js
2. `pnpm` (we recommend installing with Corepack)
3. (Optional) nx for running workspaces

When you are ready to start coding:

> 1. Start by using this template. You can do so by clicking the "Use this template" button in the top right of the GitHub UI, then "Create a new repository". Make this new repo private.

Reviewer 2: This section feels a bit disjointed, I'm wondering if it'll make more sense if we make the "process" section just these 5 bullet points, then make a new "Getting Started" section with the prereqs (nvm/pnpm/etc) and installation instructions. (And that could probably go at the very end of the README)

Author: Good callout, it definitely feels disjointed

I updated the Process section to include just those list items, and I moved everything else into a new `Getting Started` Section

Here's the Process section now:
<img width="1058" alt="Screenshot 2025-03-20 at 4 17 13 PM" src="https://github.com/user-attachments/assets/8f7f0019-8345-4afc-aefb-51ebb4f669b3" />

Here's the Getting Started section:
<img width="1071" alt="Screenshot 2025-03-20 at 4 17 28 PM" src="https://github.com/user-attachments/assets/4062dd3b-81b9-4195-9a95-040a3a719866" />
